apps/addons/tests/test_commands.py

```diff
@@ -52,3 +60,124 @@ def is_forced(ids, force):
         assert force
     monkeypatch.setattr(SIGN_ADDONS, is_forced)
     call_command('sign_addons', 123, force=True)
+
+
+# Test the "approve_addons" command.
+
+@pytest.mark.django_db
+def test_approve_addons_get_files_incomplete():
+    """An incomplete add-on can't be approved."""
+    addon = amo.tests.addon_factory(status=amo.STATUS_NULL)
+    assert approve_addons.get_files([addon.guid]) == []
+
+
+@pytest.mark.django_db
+def test_approve_addons_get_files_bad_guid():
+    """An add-on with another guid doesn't get approved."""
+    addon1 = amo.tests.addon_factory(status=amo.STATUS_UNREVIEWED, guid='foo')
+    addon1_file = addon1.latest_version.files.get()
+    addon1_file.update(status=amo.STATUS_UNREVIEWED)
+    # Create another add-on that we won't get the files for.
+    addon2 = amo.tests.addon_factory(status=amo.STATUS_UNREVIEWED, guid='bar')
+    addon2_file = addon2.latest_version.files.get()
+    addon2_file.update(status=amo.STATUS_UNREVIEWED)
+    # There's only the addon1's file returned, no other.
+    assert approve_addons.get_files(['foo']) == [addon1_file]
+
+
+@pytest.fixture(
+    params=[(amo.STATUS_UNREVIEWED, amo.STATUS_UNREVIEWED, 'prelim'),
+            (amo.STATUS_LITE, amo.STATUS_UNREVIEWED, 'prelim'),
+            (amo.STATUS_NOMINATED, amo.STATUS_NOMINATED, 'full'),
+            (amo.STATUS_PUBLIC, amo.STATUS_NOMINATED, 'full'),
+            (amo.STATUS_LITE_AND_NOMINATED, amo.STATUS_LITE, 'full')],
+    ids=['1-1-prelim',
+         '8-1-prelim',
+         '3-3-full',
+         '4-3-full',
+         '9-8-full'])
+def use_case(request, db):
+    """This fixture will return quadruples for different use cases.
+
+    Addon                   | File1 and 2        | Review type
+    ==============================================================
+    waiting for prelim      | waiting for prelim | prelim reviewed
+    prelim reviewed         | waiting for prelim | prelim reviewed
+    waiting for full        | waiting for full   | fully reviewed
+    fully reviewed          | waiting for full   | fully reviewed
+    prelim waiting for full | prelim reviewed    | fully reviewed
+    """
+    addon_status, file_status, review_type = request.param
+
+    # Create the "mozilla" user which is used to log the approval.
+    amo.tests.user_factory(id=settings.TASK_USER_ID)
+
+    addon = amo.tests.addon_factory(status=addon_status, guid='foo')
+    version = addon.latest_version
+    file1 = version.files.get()
+    file1.update(status=file_status)
+    # A second file for good measure.
+    file2 = amo.tests.file_factory(version=version, status=file_status)
+    # If the addon is public, and we change its only file to something else
+    # than public, it'll change to unreviewed.
+    addon.update(status=addon_status)
+    assert addon.reload().status == addon_status
+    assert file1.reload().status == file_status
+    assert file2.reload().status == file_status
+
+    return (addon, file1, file2, review_type)
```

Comment:
This "parametrized pytest fixture" will result in the tests using it to be run once for each "use case". This will result in the following:

<img width="1020" alt="screen shot 2015-09-03 at 18 55 42" src="https://cloud.githubusercontent.com/assets/167767/9665014/b7f1de76-526d-11e5-8647-0b9bddee7b93.png">